js/script.js

```diff
@@ -224,7 +263,20 @@ const handlePagination = () => {
   renderTasks(currentFilter, true);
 };
 
+const handleSearchIconClick = () => {
+  if (
+    $searchInput.style.display === "none" ||
+    $searchInput.style.display === ""
+  ) {
+    $searchInput.style.display = "block";
+    $searchInput.focus();
+  } else {
+    $searchInput.style.display = "none";
+  }
+};
```

Comment:
Please we declare a `show` with `display: block` property and another one `hide` with `display:none` and use them to implement the logic for them. You had to deal with display property in many cases. So it's better to declare a class and use it as per your requirements.